docs/development/design/3-DN-Retrieving_unique_graphs_from_Otel_data/class_diagrams.puml

```diff
@@ -8,35 +8,30 @@ struct OTelEvent {
     start_timestamp: str
     end_timestamp: str
     application_name: str
-    parent_event_id: str
+    parent_event_id: Optional[str]
     child_event_ids: Optional[list[str], None]
 }
 
 abstract class OTELDataSource {
-    #yaml_config: dict[str, Any]
-    #valid_data_sources: list[str]
-    #data_source: str
-    +__init__(): None
     +__iter__(): Self
     +__next__(): OTelEvent
-    +get_yaml_config(): Any
-    +set_file_ext(): str
 }
 
 class JSONDataSource {
+    #config: JSONDataSourceConfig
```

Comment:
Should we maybe add to the class diagram this class and its dependencies as we are mentioning it?

Reply:
Yes good point